core/templates/domain/summary/learner-exploration-summary.model.ts

```diff
@@ -51,7 +51,7 @@ export interface LearnerExplorationSummaryBackendDict {
 
 export class LearnerExplorationSummary {
   constructor(
-      public category: string,
+      public category: string | undefined,
```

Comment:
This came from `collection-node.model.spec.ts` file.See line `66` we are trying to `delete summaryObject.category;` this gives error ` The operand of a 'delete' operator must be optional` so I made `category` to be an optional property.Due to this I made it `string | undefined`.
Either we have to remove that `delete` statement or introduce these changes.I checked coverage reports after removing that statement 
![image](https://user-images.githubusercontent.com/60519303/115750208-8c1c3900-a3b5-11eb-8620-9fa91a4d3b97.png)
I think removing `delete` would be better?

Reply:
Basically, the rule is: Do not modify any type only because of the spec files, in this case the spec files should always be modify to conform the already existing types.

Reply:
done